spec/factories/whitehall_import_factory.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :whitehall_import do
+    state { "importing" }
+    payload { build(:whitehall_export_document) }
+
+    after(:build) do |record|
+      record.whitehall_document_id = record.payload["id"]
+      record.content_id = record.payload["content_id"]
+    end
```

Comment:
You don't actually need these to be in an `after(:build)` you can instead do:

```
    whitehall_document_id { payload["id"] }
    content_id { payload["content_id"] }
```

Reply:
Ahh that's nicer, thanks!